src/node/routes/login.ts

```diff
@@ -98,4 +88,35 @@ router.post("/", async (req, res) => {
   } catch (error) {
     res.send(await getRoot(req, error))
   }
+}
+
+export const router = Router()
+
+router.use((req : any, res : any, next : any) => {
+  const to = (typeof req.query.to === "string" && req.query.to) || "/"
+  if (authenticated(req)) {
+    return redirect(req, res, to, {
+      to: undefined, 
+      password: undefined,
+      pass: undefined,
+    })
+  }
+  next()
 })
+
+router.get("/", async (req : any, res : any) => {
+  if (req.args["enable-get-requests"]) {
+    // `?password` overrides `?pass`
+    if (req.query.password) {
+      return await login(req, res, req.query.password as string)
```

Comment:
The query variables here can be objects or arrays so it might be safer to check if it's the right type instead of typecasting.

```suggestion
    if (req.query.password && typeof req.query.password === "string") {
      return await login(req, res, req.query.password)
```

Reply:
Didn't think of that thanks!